pkg/kubeclient/kubelet_client.go

```diff
@@ -155,9 +155,14 @@ func (client *KubeletClient) callKubeletEndpoint(ip, path string) ([]byte, error
 		return nil, err
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), defaultConnTimeOut)
```

Comment:
Nit - As this technically isn't a connection timeout, let's not overload it?  We still can set it to the same 20 seconds, but let's use a different name?